packages/dev/core/src/Culling/Helper/transformFeedbackBoundingHelper.ts

```diff
@@ -107,55 +91,49 @@ export class TransformFeedbackBoundingHelper implements IBoundingInfoHelperPlatf
                 numInfluencers = manager.numMaxInfluencers || manager.numInfluencers;
                 if (numInfluencers > 0) {
                     defines.push("#define MORPHTARGETS");
-                    defines.push("#define MORPHTARGETS_POSITION");
-                }
-                if (manager.isUsingTextureForTargets) {
-                    defines.push("#define MORPHTARGETS_TEXTURE");
-
-                    if (uniforms.indexOf("morphTargetTextureIndices") === -1) {
-                        uniforms.push("morphTargetTextureIndices");
+                    if (manager.hasPositions) defines.push("#define MORPHTARGETTEXTURE_HASPOSITIONS");

```

Comment:
We have this type of code 7 or 8 times, I think we should try to factor it.

Instead of modifying `PrepareAttributesForMorphTargetsInfluencers`, you could create a new `PrepareDefinesAndAttributesForMorphTargets` function that would be something like this (warning: not tested!):

```typescript
function PrepareDefinesAndAttributesForMorphTargets(
    morphTargetManager: MorphTargetManager,
    defines: string[],
    attribs: string[],
    mesh: AbstractMesh,
    usePositionMorph = true,
    useNormalMorph = false,
    useTangentMorph = false,
    useUVMorph = false,
    useUV2Morph = false
): void {
    const numMorphInfluencers = morphTargetManager.numMaxInfluencers || morphTargetManager.numInfluencers;
    if (numMorphInfluencers <= 0) {
        return;
    }

    defines.push("#define MORPHTARGETS");

    if (morphTargetManager.hasPositions) defines.push("#define MORPHTARGETTEXTURE_HASPOSITIONS");
    if (morphTargetManager.hasNormals) defines.push("#define MORPHTARGETTEXTURE_HASNORMALS");
    if (morphTargetManager.hasTangents) defines.push("#define MORPHTARGETTEXTURE_HASTANGENTS");
    if (morphTargetManager.hasUVs) defines.push("#define MORPHTARGETTEXTURE_HASUVS");
    if (morphTargetManager.hasUV2s) defines.push("#define MORPHTARGETTEXTURE_HASUV2S");

    if (morphTargetManager.supportsPositions && usePositionMorph) defines.push("#define MORPHTARGETS_POSITION");
    if (morphTargetManager.supportsNormals && useNormalMorph) defines.push("#define MORPHTARGETS_NORMAL");
    if (morphTargetManager.supportsTangents && useTangentMorph) defines.push("#define MORPHTARGETS_TANGENT");
    if (morphTargetManager.supportsUVs && useUVMorph) defines.push("#define MORPHTARGETS_UV");
    if (morphTargetManager.supportsUV2s && useUV2Morph) defines.push("#define MORPHTARGETS_UV2");

    defines.push("#define NUM_MORPH_INFLUENCERS " + numMorphInfluencers);

    if (morphTargetManager.isUsingTextureForTargets) {
        defines.push("#define MORPHTARGETS_TEXTURE");
    }

    _TmpMorphInfluencers.NUM_MORPH_INFLUENCERS = numMorphInfluencers;
    _TmpMorphInfluencers.NORMAL = useNormalMorph;
    _TmpMorphInfluencers.TANGENT = useTangentMorph;
    _TmpMorphInfluencers.UV = useUVMorph;
    _TmpMorphInfluencers.UV2 = useUV2Morph;

    PrepareAttributesForMorphTargets(attribs, mesh, _TmpMorphInfluencers, usePositionMorph);
}
```
With this function, you can replace all the boilerplate code with:
```typescript
if (manager) {
    PrepareDefinesAndAttributesForMorphTargets(manager, defines, attribs, mesh, ...);
}
```

Reply:
I realized during refactoring that there is no function in the materialHelper.functions file that handles string[] type defines. They all handle the case of using the MaterialDefines class.
For this reason, it seems inconsistent to add a function to this file that handles string[] define. This is actually why I didn't add the new helper function.

I think we need a refactoring to unify the way we handle the defines first, so we can reuse the same prepare code.

Anyway, I'll make the change as requested.

Reply:
I don't think it's a problem to add the function in `materialHelper.functions`, but I'll let @sebavan do the final review when he comes back (early next week).

Thanks a lot for the PR!